server/bin/pbench-index.py

```diff
@@ -181,344 +135,15 @@ def main(options, name):
-                    rename_tb_link(
-                        tb, Path(controller_path, linkerrdest), idxctx.logger
-                    )
-                idxctx.logger.info("Finished {} (size {:d})", tb, size)
-        except Exception:
-            idxctx.logger.exception("Unexpected setup error")
-            res = 12
-        else:
-            # No exceptions while processing tar ball, success.
-            res = 0
-        finally:
-            if idxctx:
-                idxctx.dump_opctx()
-            idxctx.logger.debug("stopped processing list of tar balls")
-
-            idx = _count_lines(indexed)
-            skp = _count_lines(skipped)
-            err = _count_lines(erred)
-
-            idxctx.logger.info(
-                "{}.{}: indexed {:d} (skipped {:d}) results," " {:d} errors",
-                name,
-                idxctx.TS,
-                idx,
-                skp,
-                err,
-            )
-
-            if err > 0:
-                if skp > 0:
-                    subj = (
-                        f"{name}.{idxctx.TS} - Indexed {idx:d} results, skipped {skp:d}"
-                        f" results, w/ {err:d} errors"
-                    )
-                else:
-                    subj = (
-                        f"{name}.{idxctx.TS} - Indexed {idx:d} results, w/ {err:d}"
-                        " errors"
-                    )
-            else:
-                if skp > 0:
-                    subj = f"{name}.{idxctx.TS} - Indexed {idx:d} results, skipped {skp:d} results"
-                else:
-                    subj = f"{name}.{idxctx.TS} - Indexed {idx:d} results"
+    idx = Index(name, options, idxctx, INCOMING_rp)
+    status, value = idx.collect_tb(ARCHIVE_rp, qdir)
 
-            report_fname = Path(tmpdir, f"{name}.{idxctx.TS}.report")
-            with report_fname.open(mode="w") as fp:
-                print(subj, file=fp)
-                if idx > 0:
-                    print("\nIndexed Results\n===============", file=fp)
-                    with indexed.open() as ifp:
-                        for line in sorted(ifp):
-                            print(line.strip(), file=fp)
-                if err > 0:
-                    print(
-                        "\nResults producing errors" "\n========================",
-                        file=fp,
-                    )
-                    with erred.open() as efp:
-                        for line in sorted(efp):
-                            print(line.strip(), file=fp)
-                if skp > 0:
-                    print("\nSkipped Results\n===============", file=fp)
-                    with skipped.open() as sfp:
-                        for line in sorted(sfp):
-                            print(line.strip(), file=fp)
-            try:
-                report.post_status(tstos(idxctx.time()), "status", report_fname)
-            except Exception:
-                pass
+    if not status:
+        return idx.process_tb(value)
```

Comment:
Here it makes sense to have `value` renamed to be `tarballs`.  And instead of `not status`, consider being more explicit with `status == 0` since that is the success value returned in `status`.

Further, instead of having two `return` statements, and introducing the bug below where `value` is returned incorrectly, why not have the return value of `process_tb(tarballs)` assigned to `status` and just `return status` below?